server/block/leaves.go

```diff
@@ -66,6 +67,14 @@ func (l Leaves) RandomTick(pos cube.Pos, w *world.World, _ *rand.Rand) {
 			l.ShouldUpdate = false
 			w.SetBlock(pos, l, nil)
 		} else {
+			ctx := event.C()
+			if w.Handler().HandleLeavesDecay(ctx, pos); !ctx.Cancelled() {
+				drops := l.BreakInfo().Drops(nil, nil)
+				for _, drop := range drops {
+					dropItem(w, drop, pos.Vec3Centre())
+				}
+			}
+
 			w.SetBlock(pos, nil, nil)
```

Comment:
Should it not only set the block to nil if the event was not cancelled? I think the logic of only dropping items would be slightly unexpected?